sway-fmt/src/traversal_helper.rs

```diff
@@ -85,11 +85,47 @@ pub fn format_delineated_path(line: &str) -> String {
     line.chars().filter(|c| !c.is_whitespace()).collect()
 }
 
+// Same as s.match_indices(|ch| {}) but allows to match by checking &str vs char
+fn match_indices_str(s: &str) -> Vec<(usize, &str)> {
+    let mut res: Vec<(usize, &str)> = Vec::new();
+
+    // Match the as token with spaces so as to avoid imprperly matching an 'as' substring
+    // in another type of token
+    let as_token = " as ";
+    let mut start = 0;
+
+    while start < s.len() {
+        // Try to match the 'as' token first then fallback to single chars
+        if start <= s.len() - 4 && &s[start..start + as_token.len()] == as_token {
```

Comment:
Can `s.len()` ever be less than `4`? If so, I think the `s.len() - 4` expression may underflow and `panic!` or wrap around as a result.

Reply:
Also, where does `4` come from? Can we get this from a const or retrieving the length directly from what this refers to?

Reply:
I can make this change 👍